blockchain/chain_sync/src/sync.rs

```diff
@@ -193,6 +194,38 @@ where
                             )
                             .await
                     }
+                    Some(NetworkEvent::PubsubMessage { source, message }) => {
+                        match message {
+                            forest_libp2p::PubsubMessage::Block(b) => {
+                                info!("Received block over GossipSub: {} from {}", b.header.epoch(), source.clone().unwrap());
+                                // Get bls_messages in the store or over Bitswap
+                                let mut bmsgs: Vec<UnsignedMessage> = Vec::with_capacity(b.bls_messages.len());
+                                for m in b.bls_messages.clone() {
+                                    let bmsg: UnsignedMessage = self.network.bitswap_get(m).await.unwrap();
+                                    bmsgs.push(bmsg);
+                                }
+                                // Get secp_messages in the store or over Bitswap
+                                let mut smsgs: Vec<SignedMessage> = Vec::with_capacity(b.secpk_messages.len());
+                                for m in b.secpk_messages.clone() {
+                                    let smsg: SignedMessage = self.network.bitswap_get(m).await.unwrap();
+                                    smsgs.push(smsg);
+                                }
+                                // Form block
+                                let block = Block {
+                                    header: b.header,
+                                    bls_messages: bmsgs,
+                                    secp_messages: smsgs,
+                                };
+                                let ts = FullTipset::new(vec![block]).unwrap();
+                                if let Err(e) = self.inform_new_head(source.clone().unwrap(), &ts).await {
```

Comment:
the source isn't guaranteed because I don't believe we reject responses with no peer id set, should handle this in some way

Reply:
The source address should be guaranteed because of the signing policies we enforce on gossipsub messages.

Reply:
no, actually we explicitly allow this (because with out it before it was an issue) If you want to ensure this is the case, you need to change the `ValidationMode` on the GossipSub config. In any case, it would be much safer if there wasn't this unwrap when dealing with inputs directly from the network as it's too easy to trigger

Reply:
I've changed the validation mode to Strict and ive tested it against a local lotus devnet.